localization/ndt_scan_matcher/include/ndt_scan_matcher/ndt_scan_matcher_core.hpp

```diff
@@ -76,32 +76,23 @@ struct NdtResult
   std::vector<geometry_msgs::msg::Pose> transformation_array;
 };
 
-template <typename PointSource, typename PointTarget>
-std::shared_ptr<NormalDistributionsTransformBase<PointSource, PointTarget>> get_ndt(
-  const NDTImplementType & ndt_mode)
+struct NDTParams
 {
-  std::shared_ptr<NormalDistributionsTransformBase<PointSource, PointTarget>> ndt_ptr;
-  if (ndt_mode == NDTImplementType::PCL_GENERIC) {
-    ndt_ptr.reset(new NormalDistributionsTransformPCLGeneric<PointSource, PointTarget>);
-    return ndt_ptr;
-  }
-  if (ndt_mode == NDTImplementType::PCL_MODIFIED) {
-    ndt_ptr.reset(new NormalDistributionsTransformPCLModified<PointSource, PointTarget>);
-    return ndt_ptr;
-  }
-  if (ndt_mode == NDTImplementType::OMP) {
-    ndt_ptr.reset(new NormalDistributionsTransformOMP<PointSource, PointTarget>);
-    return ndt_ptr;
-  }
-
-  const std::string s = fmt::format("Unknown NDT type {}", static_cast<int>(ndt_mode));
-  throw std::runtime_error(s);
-}
+  double trans_epsilon;
+  double step_size;
+  double resolution;
+  int max_iterations;
+  pclomp::NeighborSearchMethod search_method;
+  int num_threads;
+  float regularization_scale_factor;
+};
 
 class NDTScanMatcher : public rclcpp::Node
 {
   using PointSource = pcl::PointXYZ;
   using PointTarget = pcl::PointXYZ;
+  using NormalDistributionsTransform =
+    pclomp::NormalDistributionsTransform<PointSource, PointTarget>;
 
   // TODO(Tier IV): move file
   struct OMPParams
```

Comment:
I think we no longer need a OMPParams.

Reply:
Thanks for all the comments!

For `converged_param` comments: https://github.com/autowarefoundation/autoware.universe/pull/2053/commits/9ae2705cb7f47f84ab250401dcbfe5734529806c
For OMPParams comment: https://github.com/autowarefoundation/autoware.universe/pull/2053/commits/0248009f64b6efcf2c1fe96b3a533759f8f06eaa